src/groupByKey.ts

```diff
@@ -1,7 +1,20 @@
 interface Item {
-  [key: string]: any;
+  [key: string]: any,
 }
 
-export function groupByKey(items, field) {
-  // write code here;
+export function groupByKey<T extends Item, K extends keyof T>(
+  items: T[],
+  field: K,
+):
+  Record<T[K], T[]> {
+  return items.reduce(
+    (accumulator, item) => {
+      const value = item[field];
+
+      accumulator[value] = (accumulator[value] || []).concat(item);
+
+      return accumulator;
+    },
+    {} as Record<T[K], T[]>
+  );
```

Comment:
Don't overuse generic types. Use them only when necessary. 
Here it would be much better just to introduce additional type, like this:
```
type ItemsGroup<T> = {
  [key: string]: T[]
};
```
And then use it:
```suggestion
export function groupByKey(
  items: Item[],
  field: keyof Item,
): ItemGroup<Item>{
  return items.reduce(
    (accumulator, item) => {
      const value = item[field];

      accumulator[value] = (accumulator[value] || []).concat(item);

      return accumulator;
    },
    {},
  );
```